src/Uno.UWP/ApplicationModel/Contacts/ContactManager.Android.cs

```diff
@@ -0,0 +1,100 @@
+﻿#if __ANDROID__
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Windows.Foundation;
+
+namespace Windows.ApplicationModel.Contacts
+{
+	public partial class ContactManager
+	{
+
+		public static IAsyncOperation<ContactStore> RequestStoreAsync() => RequestStoreAsync(ContactStoreAccessType.AllContactsReadOnly);
+
+		public static IAsyncOperation<ContactStore> RequestStoreAsync(ContactStoreAccessType accessType) => RequestStoreAsyncTask(accessType).AsAsyncOperation<ContactStore>();
+
+		private static async Task<ContactStore> RequestStoreAsyncTask(ContactStoreAccessType accessType)
+		{
+			// UWP: AppContactsReadWrite, AllContactsReadOnly, AllContactsReadWrite (cannot be used without special provisioning by Microsoft)
+			// Android: Manifest has READ_CONTACTS and WRITE_CONTACTS, no difference between app/limited/full
+			// using only AllContactsReadOnly, other: throw NotImplementedException
+
+			if (accessType != ContactStoreAccessType.AllContactsReadOnly)
+				throw new NotImplementedException();
+
+			// do we have declared this permission in Manifest?
+			// it could be also Coarse, without GPS
+			Android.Content.Context context = Android.App.Application.Context;
+			Android.Content.PM.PackageInfo packageInfo =
+				context.PackageManager.GetPackageInfo(context.PackageName, Android.Content.PM.PackageInfoFlags.Permissions);
+			var requestedPermissions = packageInfo?.RequestedPermissions;
+			if (requestedPermissions is null)
+				return null;

```

Comment:
Is null what the windows API returns ?

Reply:
yes

Reply:
(just tested)

Reply:
Sorry. I tested, null is returned when user denies access.
But if there is no "Contacts" defined in Manifest, it throws System.UnauthorizedAccessException.
I would change it in next commit - but I don't want to make one commit per change; make batch of changes (from all files, and from all answers).